src/scripts/clipperUI/ratingsHelper.ts

```diff
@@ -0,0 +1,538 @@
+							});
+						}
+					}, {
+						id: Constants.Ids.ratingsButtonFeedbackNo,
+						label: Localization.getLocalizedString("WebClipper.Label.Ratings.Button.NoThanks"),
+						handler: () => {
+							panel.setState({
+								userSelectedRatingsPromptStage: RatingsPromptStage.NONE
+							});
+						}
+					});
+				} else {
+					// this shouldn't happen
+					panel.setState({
+						userSelectedRatingsPromptStage: RatingsPromptStage.NONE
+					});
+				}
+				break;
+			default:
+			case RatingsPromptStage.END:
+			case RatingsPromptStage.NONE:
+				break;
+		}
+
+		return buttons;
+	}
+
+	/**
+	 * We will show the ratings prompt if ALL of the below applies:
+	 *   * Ratings prompt is enabled for the ClientType/ClipperType
+	 *   * If StorageKeys.doNotPromptRatings is not "true"
+	 *   * If RatingsHelper.badRatingTimingDelayIsOver(...) returns true when provided StorageKeys.lastBadRatingDate
+	 *   * If RatingsHelper.badRatingVersionDelayIsOver(...) returns true when provided StorageKeys.lastBadRatingVersion and StorageKeys.lastSeenVersion
+	 *   * If RatingsHelper.clipSuccessDelayIsOver(...) returns true when provided StorageKeys.numClipSuccess
+	 */
+	public static shouldShowRatingsPrompt(clipperState: ClipperState): Promise<boolean> {
+		return new Promise<boolean>((loggingResolve, loggingReject) => {
+			let shouldShowRatingsPromptEvent = new Log.Event.PromiseEvent(Log.Event.Label.ShouldShowRatingsPrompt);
+			let shouldShowRatingsPromptInfo: RatingsLoggingInfo = { shouldShowRatingsPrompt: undefined };
+
+			RatingsHelper.shouldShowRatingsPromptInternal(clipperState, shouldShowRatingsPromptEvent, shouldShowRatingsPromptInfo).then((shouldShowRatingsPrompt: boolean) => {
+				shouldShowRatingsPromptInfo.shouldShowRatingsPrompt = shouldShowRatingsPrompt;
+				loggingResolve(shouldShowRatingsPrompt);
+			}, () => {
+				shouldShowRatingsPromptInfo.shouldShowRatingsPrompt = false;
+				loggingReject(undefined);
+			}).then(() => {
+				shouldShowRatingsPromptEvent.setCustomProperty(Log.PropertyName.Custom.RatingsInfo, JSON.stringify(shouldShowRatingsPromptInfo));
+				Clipper.logger.logEvent(shouldShowRatingsPromptEvent);
+			});
+		});
+	}
+
+	/**
+	 * Get the animation strategy for the ratings subpanel of the success panel provided
+	 */
+	public static getAnimationStategy(panel: SuccessPanelClass): AnimationStrategy {
+		return new SlideFromRightAnimationStrategy({
+			extShouldAnimateIn: () => {	return panel.state.userSelectedRatingsPromptStage !== panel.state.currentRatingsPromptStage; },
+			extShouldAnimateOut: () => { return false; },
+			onAfterAnimateIn: () => { panel.setState({ currentRatingsPromptStage: panel.state.userSelectedRatingsPromptStage }); }
+		});
+	}
+
+	/**
+	 * Adds 1 to the value in StorageKeys.numClipSuccess.
+	 * If the value does not exist yet or is invalid, we (re)initialize it to 1.
+	 */
+	public static incrementClipSuccessCount(): Promise<void> {
+		return new Promise<void>((resolve, reject) => {
```

Comment:
This function and behavior could belong in `saveToOneNote::startClip(...)`? I remember implementing something like this when I did tooltips and I think it looked identical LOL (but I ended up scrapping it :stuck_out_tongue_closed_eyes: